server/chat-plugins/github.ts

```diff
@@ -0,0 +1,262 @@
+/**
+ * Chat plugin to view GitHub events in a chatroom.
+ * By Mia, with design / html from xfix's original bot, https://github.com/xfix/GitHub-Bot-Legacy/
+ * @author mia-pi-git
+ */
+
+import {FS, Utils} from '../../lib';
+
+const STAFF_REPOS = Config.staffrepos || ['pokemon-showdown', 'pokemon-showdown-client', 'Pokemon-Showdown-Dex'];
+const COOLDOWN = 10 * 60 * 1000;
+
+export const gitData: GitData = JSON.parse(FS("config/chat-plugins/github.json").readIfExistsSync() || "{}");
+
+interface GitHookHandler {
+	on(event: 'pull_request', callback: (repo: string, ref: string | undefined, result: PullRequest) => void): void;
+	on(event: 'push', callback: (repo: string, ref: string, result: Push) => void): void;
+	on(event: string, callback: (repo: string, ref: string | undefined, result: any) => void): void;
+	listen(): void;
+	server: import('http').Server;
+}
+
+interface Push {
+	commits: Commit[];
+	sender: {login: string};
+	compare: string;
+}
+
+interface PullRequest {
+	action: string;
+	number: number;
+	pull_request: {
+		url: string,
+		html_url: string,
+		title: string,
+		user: {
+			login: string,
+			html_url: string,
+		},
+		merge_commit_sha: string,
+	};
+	sender: {login: string};
+}
+
+interface Commit {
+	id: string;
+	message: string;
+	author: {name: string, avatar_url: string};
+	url: string;
+}
+
+interface GitData {
+	usernames?: {[username: string]: string};
+	bans?: {[username: string]: string};
+}
+
+export const GitHub = new class {
+	readonly hook: GitHookHandler | null = null;
+	updates: {[k: string]: number} = Object.create(null);
+	constructor() {
+		try {
+			this.hook = (require('githubhook'))(Config.github);
```

Comment:
It would probably be a good idea to document `Config.github` somewhere; at least link to [the documentation](https://www.npmjs.com/package/githubhook).